src/features/island/hud/LandscapingHud.tsx

```
@@ -119,6 +145,66 @@ const LandscapingHudComponent: React.FC<{ location: PlaceableLocation }> = ({
+    } finally {
+      setSaving(false);
+    }
+  };
+
+  // The floor buttons go through switchSurface, but browser back or a typed
+  // URL can land the player on another surface with landscaping still
+  // active and the draft still keyed to the one they left. Settle it the same
+  // way: save that surface, then continue here.
+  useEffect(() => {
+    if (landscapingLocation && landscapingLocation !== location) {
+      void switchSurface(location);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/features/island/hud/LandscapingHud.tsx:203:12
  201 |   useEffect(() => {
  202 |     if (landscapingLocation && landscapingLocation !== location) {
> 203 |       void switchSurface(location);
      |            ^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  204 |     }
  205 |     // Mount only: this is about how the player arrived, not later changes.
  206 |     // eslint-disable-next-line react-hooks/exhaustive-deps

(react-hooks/set-state-in-effect)